components/ui/navigation/window-controls.tsx

```
@@ -11,55 +11,65 @@ interface WindowControlsProps {
   return (
-    <div className={`flex items-center space-x-2 mr-4 ${className}`}>
+    <div className={`flex items-center ${spacingClass} ${marginClass} ${className}`}>
       {/* Close Button */}
       <button
         aria-label="Close"
         title="Close"
-        className="relative group w-3 h-3 rounded-full bg-red-500 hover:bg-red-600 transition-colors flex items-center justify-center"
+        className={`relative group ${buttonSize} rounded-full bg-red-500 hover:bg-red-600 transition-colors flex items-center justify-center`}
         onClick={onClose}
         disabled={!onClose} // Disable based solely on prop presence
       >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)